### PR TITLE
AsyncEvent refactoring final merge

### PR DESCRIPTION
**Pull request by palpatim**
*Description of changes:*

This PR rebases https://github.com/aws-amplify/amplify-ios/tree/palpatim/async-event-refactoring against master, and fixes a few stray issues:
- Removed unnecessary AWSDataStoreCategoryPluginTests.xcscheme
- Fix reference to awsconfiguration.json in AWSAuthPluginIntegrationTests
- Fix AWSAuthPluginTests formatting & linting
- Fix corrupted project file reference to AWSAuthSignUpOperation.swift
- Fix Analytics formatting & linting
- Change AuthDirective tests to use OperationResult instead of AsyncEvent
- Fix linting & formatting
- Fix corrupted project file reference to AuthResetPasswordRequest.swift

And importantly:
- Update CHANGELOG & prep for release

Although the base of this PR is https://github.com/aws-amplify/amplify-ios/tree/palpatim/async-event-refactoring, that is only to help isolate the changes for review. Once the PR is reviewed, I will change the base to https://github.com/aws-amplify/amplify-ios/tree/master and merge it. Therefore, it is safe to disregard conflicts with the "async-refactor" branch.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,9 @@ xcuserdata
 build
 builtFramework
 AWS*.framework
+node_modules
+package.json
+package-lock.json
```
Comment (lawmicha) [inline .gitignore:33]: i wonder where these came from, i also see the package.json file added in these commits

File: AmplifyPlugins/Analytics/AWSPinpointAnalyticsPluginIntegrationTests/AWSPinpointAnalyticsPluginTestBase.swift [hunk 2]
```diff
@@ -5,13 +5,12 @@
...
-import AWSPinpointAnalyticsPlugin
 import AWSMobileClient
+import AWSPinpointAnalyticsPlugin
+import XCTest
 
 class AWSPinpointAnalyticsPluginTestBase: XCTestCase {
-
     let appId: JSONValue = "56e6f06fd4f244c6b202bc327bd3b4e6"
     let region: JSONValue = "us-east-1"
     let targetingRegion: JSONValue = "us-east-1"
```
Comment (lawmicha) [inline AmplifyPlugins/Analytics/AWSPinpointAnalyticsPluginIntegrationTests/AWSPinpointAnalyticsPluginTestBase.swift:16]: opening issue for self to address, fix Analytics integration test 
- move config over to configuration file + readme for provisioning
- remove AWSMobileClient dependency for getting identityId

File: package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/aws-amplify/amplify-ios/issues"
+  },
+  "homepage": "https://github.com/aws-amplify/amplify-ios#readme",
+  "devDependencies": {
+    "standard-version": "^8.0.0"
+  }
+}
```
Comment (lawmicha) [inline package.json:23]: i wonder what this is used for? i'm assuming this change is coming from a merge from some other commit

File: .gitignore (hunk 1 above)
Comment (palpatim) [reply .gitignore:33]: Ack! I didn't mean to add the package.json file--I'll remove it.

I added this explicitly so we can use NPM-based build tools (like [standard-version](https://www.npmjs.com/package/standard-version)) to help with CI/CD. We may end up refactoring that out.

File: package.json (hunk 3 above)
Comment (palpatim) [reply package.json]: Removed

**Pull request merged**